### DISCUSSION TITLE
MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version

### DISCUSSION THREAD

**Discussion by dkbarn** [sqlalchemy/sqlalchemy#10489] [Usage Questions]
I am getting a load of these messages printed to the terminal when running sqlalchemy with mysqlclient. According to the discussion [here](https://github.com/PyMySQL/mysqlclient/discussions/651#discussioncomment-6821149) and [here](https://github.com/PyMySQL/mysqlclient/discussions/648#discussioncomment-6759502) this is caused by issuing `conn.ping(True)` or `conn.ping(False)`, which I can see that sqlalchemy is doing here: https://github.com/sqlalchemy/sqlalchemy/blob/efe4025c068d6746637db362245ab88ce7dfe785/lib/sqlalchemy/dialects/mysql/mysqldb.py#L171

Would it be possible to remove the argument to `ping` to resolve this deprecation warning?

See also:
https://bugs.mysql.com/bug.php?id=112089

**Comment by zzzeek** [comment]
we can try, but looking at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/809/ this patch started without the False five years ago and we needed to add it in order to pass the test suite.

**Reply by zzzeek** [reply]
negatory.  this option defaults to True and indicates that ping should silently reconnect to the DB.  We need it to not do that.    I dont see any docs for the mysqlclient version of this so I would need confirmation from the maintainer that the flag defaults to False in any case (which I dont know that this is true).

tests fail for pymysql where this flag defaults to True: https://pymysql.readthedocs.io/en/latest/modules/connections.html#pymysql.connections.Connection.ping

**Reply by zzzeek** [reply]
OK confirmed we have to open an issue for this

**Reply by zzzeek** [reply]
#10492
